documentation/CAMARA-API-access-and-user-consent.md

```diff
@@ -354,3 +358,52 @@ If some usecase/s for an API point to off-net scenarios and where consumption an
     - Device connected to WiFi
     - Device without UI (IoT)
 
+## CAMARA API Specification - Authorization and authentication common guidelines
+
+The purpose of this document section is to standardise the specification of `securitySchemes` and `security` across all CAMARA API subprojects with common mandatory guidelines as agreed by the Technical Steering Committee (TSC) and the participants of this Working Group.
+
+### Use of openIdConnect for `securitySchemes`
+
+In general, OpenID Connect is the protocol to be used for securitization. Each API specification must ONLY define the following openIdConnect entry in `securitySchemes`, as shown in document [CAMARA-AuthN-AuthZ-Concept.md](https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/CAMARA-AuthN-AuthZ-Concept.md#documentation-and-specs-):
+
+```
+components:
+  securitySchemes:
+    openId:
+      type: openIdConnect
+      openIdConnectUrl: https://example.com/.well-known/openid-configuration
+```
+
+The value for `openIdConnectUrl` in the CAMARA spec is an example, that must be substituted by the specific discovery endpoint for OIDC protocol of the API provider, when the API is exposed in one of its environments.
+
+### Use of `security` property
+
+Generally, each operation is protected by a scope and it will include a security property with a single element in the array:
+
+```
+paths:
+  {path}:
+    {method}:
+      ...
+      security:
+        - openId:
+            - {scope}
+```
+The key is arbitrary in OAS, but convention in CAMARA is to name it `openId`. This key must be same defined in the `components.securitySchemes` section.
+
+The {scope} is the specific scope defined to protect this operation.
+
+### Mandatory template for `info.description` in CAMARA API specs
+
+The documentation template below must be used as part of the API documentation in  `info.description` property in the CAMARA API specs:
+
+
+```
+### Authorization and authentication
+
+CAMARA guidelines defines a set of authorization flows which can grant API clients access to the API functionality, as outlined in the document [CAMARA-API-access-and-user-consent.md](https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/CAMARA-API-access-and-user-consent.md). Which specific authorization flows are to be used will be determined during onboarding process, happening between the API Client and the Telco Operator exposing the API, taking into account the declared purpose for accessing the API, while also being subject to the prevailing legal framework dictated by local legislation.
```

Comment:
Document "CAMARA-API-access-and-user-consent.md" says nothing about client credentials, and yet that remains a valid authorisation scheme for _some_ CAMARA API endpoints. When it was proposed that CAMARA adopt `type: openIdConnect` as the ONLY valid security scheme, this was justified on the basis that `.well-known/openid-configuration` allows `client_credentials` in the list of  `grant_types_supported`. And yet here we have a reference to a document that does not "outline" this as a possibility.

So for the proposed text to be applicable, "CAMARA-API-access-and-user-consent.md" must be updated to list client credentials as a valid authorisation scheme for some scenarios.

Reply:
Agree. This is a fair point. We will commit an update to clarify that client credentials is a valid authorization scheme for some scenarios.

Reply:
Done in commit https://github.com/camaraproject/IdentityAndConsentManagement/pull/93/commits/4f09ec6530261b425c560e337dd8d56dbfd15c0a